File: 0350-intersection-of-two-arrays-ii/0350-intersection-of-two-arrays-ii.py
class Solution:
    def intersect(self, nums1: List[int], nums2: List[int]) -> List[int]:
        result = []
        counter = Counter(nums1) # Counter는 요소의 빈도를 세는 데 최적화된 딕셔너리 서브클래스입니다.

        for num in nums2:
            print(num, counter)
            if counter[num] > 0:
                counter[num] -= 1
                result.append(num)
        
        return result

File: 0350-intersection-of-two-arrays-ii/0350-intersection-of-two-arrays-ii.js
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number[]}
 */
// calculate Frequency
var intersect = function(nums1, nums2) { 
    const map = new Map();
    const result = [];
    
    for (const num of nums1) {
        map.set(num, (map.get(num) || 0) + 1);
    }

    for (const num of nums2) {
        if (map.has(num) && map.get(num) > 0) {
            map.set(num, map.get(num) - 1);
            result.push(num);
        }
    }
    
    return result;
};